frontend/src/views/SecretApprovalPage/components/SecretApprovalRequest/SecretApprovalRequest.tsx

```diff
@@ -66,10 +67,19 @@ export const SecretApprovalRequest = () => {
   const { user: userSession } = useUser();
   const { permission } = useProjectPermission();
   const { data: members } = useGetWorkspaceUsers(workspaceId);
-  const isSecretApprovalScreen = Boolean(selectedApproval);
+  const isSecretApprovalScreen = Boolean(selectedApprovalId);
+  const queryParams = new URLSearchParams(window.location.search);
```

Comment:
You can use the https://nextjs.org/docs/app/api-reference/functions/use-search-params instead

Reply:
@maidul98 I believe next/navigation was added in Next 13, we're still on version 12.3.

![CleanShot 2024-08-26 at 20 52 36](https://github.com/user-attachments/assets/8edbb875-1b9f-4226-8a27-27022af1934f)

Reply:
Ohh actually useRouter has query component with it

Reply:
@akhilmhdh I wonder why we use `URLSearchParams` everywhere else? I remember we discussed this a while back, I don't remember the exact reason for using `URLSearchParams` though 🤔

Reply:
nice catch, i see it here https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object